pkg/e2e/infra/utils.go

```diff
@@ -118,6 +126,79 @@ func getMachineSetList(client runtimeclient.Client) (*mapiv1beta1.MachineSetList
 	return &machineSetList, nil
 }
 
+// getMachineSetList returns a MachineSetList object filtered by machineRoleLabel
+func getWorkerMachineSetList(client runtimeclient.Client) (*mapiv1beta1.MachineSetList, error) {
+	machineSetList := mapiv1beta1.MachineSetList{}
+	listOptions := runtimeclient.ListOptions{
+		Namespace: e2e.TestContext.MachineApiNamespace,
+	}
+
+	// See https://github.com/openshift/cluster-api-provider-kubemark/blob/master/examples/worker-machinesets.yaml#L45
+	// We plan to update the installer and consumers to rely only on applying 'node-role.kubernetes.io/worker'to machineSets
+	// in favour of 'machine.openshift.io/cluster-api-machine-role'
+	listOptions.MatchingLabels(map[string]string{nodeWorkerRoleLabel: ""})
+	if err := wait.PollImmediate(1*time.Second, time.Minute, func() (bool, error) {
+		if err := client.List(context.TODO(), &listOptions, &machineSetList); err != nil {
+			glog.Errorf("error querying api for machineSetList object: %v, retrying...", err)
+			return false, nil
+		}
+		return true, nil
+	}); err != nil {
+		glog.Errorf("Error calling getMachineSetListWorkers: %v", err)
+		return nil, err
+	}
+	if len(machineSetList.Items) > 0 {
+		return &machineSetList, nil
+	}
+
+	listOptions.MatchingLabels(map[string]string{machineRoleLabel: "worker"})
```

Comment:
I think this function should be generalised as we have identical code here. Make the generalised form takes the `labels` you want to match then create helper functions for each label type or a helper that calls the generalised form and concatenates the results.

Reply:
just getting all machineSets now